### PR TITLE
Fix flaky test by preventing duplicate keys in random mapping fields

### LINKED ISSUE #6739: [BUG] org.opensearch.client.indices.GetIndexTemplatesResponseTests.testParsingFromOpenSearchResponse is flaky
**Describe the bug**

```
OpenSearchParseException[Failed to parse content to map]; nested: JsonParseException[Duplicate field 'CZoEK'
 at [Source: (byte[])[247 bytes]; byte offset: #101]];
	at __randomizedtesting.SeedInfo.seed([3D0DAE5BAD265A1F:76B30BB90A00AF61]:0)
	at app//org.opensearch.common.xcontent.XContentHelper.convertToMap(XContentHelper.java:258)
	at app//org.opensearch.common.xcontent.XContentHelper.convertToMap(XContentHelper.java:185)
	at app//org.opensearch.common.xcontent.XContentHelper.convertToMap(XContentHelper.java:156)
	at app//org.opensearch.common.xcontent.XContentHelper.convertToMap(XContentHelper.java:141)
	at app//org.opensearch.cluster.metadata.IndexTemplateMetadata$Builder.toInnerXContent(IndexTemplateMetadata.java:425)
	at app//org.opensearch.cluster.metadata.IndexTemplateMetadata$Builder.toXContent(IndexTemplateMetadata.java:397)
	at app//org.opensearch.action.admin.indices.template.get.GetIndexTemplatesResponse.toXContent(GetIndexTemplatesResponse.java:102)
	at app//org.opensearch.client.indices.GetIndexTemplatesResponseTests.testParsingFromOpenSearchResponse(GetIndexTemplatesResponseTests.java:111)
	at java.base@19.0.2/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:104)
```

**To Reproduce**
See please https://build.ci.opensearch.org/job/gradle-check/12469/testReport/junit/org.opensearch.client.indices/GetIndexTemplatesResponseTests/testParsingFromOpenSearchResponse/

**Expected behavior**
The test should always pass

**Plugins**
Standard

**Screenshots**

**Host/Environment (please complete the following information):**
 - CI

**Additional context**

### PR DESCRIPTION
**Pull request by dbwiddis**
### Description

Although rare, [if you throw random alpha strings together often enough](https://en.wikipedia.org/wiki/Infinite_monkey_theorem), you'll get a collision.

### Related Issues

Fixes #6739

### Check List
- [x] New functionality includes testing.
  - [x] All tests pass
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dbwiddis** [comment]
Annoying when one flaky test fix has a failure from another flaky test. :) #9178

**Comment by dbwiddis** [comment]
Latest flaky test failure tracked in https://github.com/opensearch-project/OpenSearch/issues/1703...

File: client/rest-high-level/src/test/java/org/opensearch/client/indices/GetIndexTemplatesResponseTests.java [hunk 1]
```diff
@@ -99,8 +102,12 @@ public void testParsingFromOpenSearchResponse() throws IOException {
                 esIMD.settings(randomIndexSettings());
                 esIMD.putMapping("_doc", new CompressedXContent(BytesReference.bytes(randomMapping("_doc", xContentType))));
                 int numAliases = randomIntBetween(0, 8);
+                Set<String> uniqueAliases = new HashSet<>();
                 for (int j = 0; j < numAliases; j++) {
-                    esIMD.putAlias(randomAliasMetadata(String.format(Locale.ROOT, "%02d ", j) + randomAlphaOfLength(4)));
+                    uniqueAliases.add(String.format(Locale.ROOT, "%02d ", j) + randomAlphaOfLength(4));
+                }
+                for (String uniqueAlias : uniqueAliases) {
+                    esIMD.putAlias(randomAliasMetadata(uniqueAlias));
```
Comment (Poojita-Raj) [inline client/rest-high-level/src/test/java/org/opensearch/client/indices/GetIndexTemplatesResponseTests.java:110]: There doesn't seem to be a check that uniqueAliases actually has `numAliases` number of elements in it. When we add a duplicate element, the add method on line 107 should return false - a while loop check till it returns true will ensure we have the correct number of elements in uniqueAliases.

File: test/framework/src/main/java/org/opensearch/index/RandomCreateIndexGenerator.java [hunk 2]
```diff
@@ -112,8 +114,12 @@ public static void randomMappingFields(XContentBuilder builder, boolean allowObj
         builder.startObject("properties");
 
         int fieldsNo = randomIntBetween(0, 5);
+        Set<String> uniqueFields = new HashSet<>();
         for (int i = 0; i < fieldsNo; i++) {
-            builder.startObject(randomAlphaOfLength(5));
+            uniqueFields.add(randomAlphaOfLength(5));
```
Comment (Poojita-Raj) [inline test/framework/src/main/java/org/opensearch/index/RandomCreateIndexGenerator.java:119]: same here

**Review by Poojita-Raj — COMMENTED** [review]
For flaky tests, could we also mention how many times the test was run to ensure it's no longer flaky for confidence before we merge it in?

**Review by dbwiddis — COMMENTED** [review]
@Poojita-Raj I don't think the suggested changes are needed in this case.  The collisions are extremely rare and the occasional decrement is still well within (and barely impacts) the random ranges being generated.

If you really think this needs to be engineered more robustly, I'd suggest adding another utility method in `OpenSearchTestCase` similar to `generateRandomStringArray()` that creates unique strings, and then audit all the tests to see if there are other cases like this one.

> For flaky tests, could we also mention how many times the test was run to ensure it's no longer flaky for confidence before we merge it in?

I originally closed the flaky test issue because I could calculate the extremely rare probability of a failure, and asked a similar question here: https://github.com/opensearch-project/OpenSearch/issues/6739#issuecomment-1666951132

This was the only failure of this test that I could find in the last 2500 runs using the script, and a search of issues on this repo indicates it was the only case reported ever.

Happy to run this N times to prove no reproduction, but nobody has defined what N should be.   Given it's only failed once in 12500 test runs and the probability is at least 1 in 20,000 or so (probably more) I think I'd have to test it at least 30,000 times to make sure it's "fixed".   I don't think that's necessary but am willing to write a script to do so.

File: test/framework/src/main/java/org/opensearch/index/RandomCreateIndexGenerator.java (hunk 2 above)
Comment (dbwiddis) [reply test/framework/src/main/java/org/opensearch/index/RandomCreateIndexGenerator.java:119]: Simple change to a while loop preserves size.

**Review by dbwiddis — COMMENTED** [review]
@Poojita-Raj looked at it again and decided to just replace for loops with while checking size, to preserve original size. Realized the aliases were already unique because the index was used as part of the name, so backed out that change.  

Should be GTG now. I don't think it's necessary to try to reproduce this again as the cause is well understood, the reproducing duplicate keys still occur with the random seed in the linked issue, and the test doesn't fail.

**Comment by dbwiddis** [comment]
Test failure may have been fixed by https://github.com/opensearch-project/OpenSearch/pull/9079

Will rebase

**Comment by dbwiddis** [comment]
Both `CreateIndexIT` failures tracked in https://github.com/opensearch-project/OpenSearch/issues/5425

**Comment by dbwiddis** [comment]
org.opensearch.repositories.azure.AzureBlobContainerRetriesTests.testWriteBlobWithRetries https://github.com/opensearch-project/OpenSearch/issues/7934
org.opensearch.indices.replication.SegmentReplicationTargetServiceTests.testShardAlreadyReplicating https://github.com/opensearch-project/OpenSearch/issues/8928

**Comment by dbwiddis** [comment]
This has fallen off the "front page" of PR reviews, so bumping for attention. Minor fix, shouldn't be controversial.

@Poojita-Raj (answered your earlier review), @mch2 @andrross @saratvemulapalli @dblock @ryanbogan @owaiskazi19 :-)

**Review by andrross — APPROVED** [review]

**Pull request merged**